Feature/flow 3451 aes decryption

lmancel opened the pull request:
Here are the links toward the different libraries implemented for this ticket (TODO: improve description and write unit tests):
https://bitbucket.taktik.be/projects/FLOWR/repos/buffers/browse
https://bitbucket.taktik.be/projects/FLOWR/repos/udp-streamer/browse
https://bitbucket.taktik.be/projects/FLOWR/repos/ts-decryptor/browse

Commit:
feat: implement decryption pipeline

Commit:
removed TODO and obsolete interface

Commit:
fix install-widevine task

Commit:
increment travis node version to 12

hubjac1 commented on .nvmrc:
```diff
@@ -1 +1 @@
-10
+v12.14.0
```
I you have no good reason to fix a version. use latest node  12 is better.
`12`

hubjac1 replied:
Attention, travis config use node 11

hubjac1 commented:
TODO verify wexond browser

Commit:
Various stability improvements

- properly destroy all streams on close
- stop on FlowrISInitializing
- retry on stream errors
- re-evaluated player's allocated memory

Commit:
updated package-lock.json

- got to delete it and npm i for it to work

Commit:
re-evaluated memory allocation

Commit:
restore 2-steps electron install

first regular electron, then electron-widevine
otherwise there are issues in electron-builder install scripts

Commit:
fill in electron-builder's electronDownload conf

Commit:
fix(streamer): respect ffmpeg chunk structure

We were trying to split chunks into multiples of 188
However mp4 chunks are not TS chunks...

Commit:
wait for proper stream close to play

Commit:
fix: check that stored audio/subs still exist

Commit:
handle start of stream backpressure

It takes some time before ffmpeg can handle the data we send
We need to make sure to buffer enough data before sending it
- also fixed hasStreamChanged

Commit:
flush on replay + robustify sender

denishamann commented on src/frontend/src/player.ts:
```diff
@@ -27,252 +63,368 @@ export class Player {
-          this.updateChannelData(this.currentStreams)
+  async stop(): Promise<void> {
+    await this.stopping
+    this.stopping = new Promise(async (resolve) => {
```
```suggestion
    this.stopping = new Promise(async resolve => {
```

denishamann commented on src/frontend/src/player.ts:
```diff
@@ -27,252 +63,368 @@ export class Player {
+    } else {
+      input = url
+    }
+
```
`const input = this.decryption.use ? await this.getDecryptionPipeline(url) : url`

lmancel replied:
Sometimes I prefer a simple if / else statement rather than a 100 char long ternary

denishamann commented on src/frontend/src/player.ts:
```diff
@@ -27,252 +63,368 @@ export class Player {
+    }
+
+    try {
+      const metadata = await ffprobe(input, { timeout: 30 })
```
```suggestion
      return ffprobe(input, { timeout: 30 })
```

lmancel replied:
I'm trying to use a try / catch block here, if I return a Promise that fails it won't be properly caught with this.

denishamann commented on src/frontend/src/player.ts:
```diff
@@ -27,252 +63,368 @@ export class Player {
+  hasStreamChanged(newStreamData: ICurrentStreams, localCurrentStream: ICurrentStreams | undefined): boolean {
+    // If current and new streams first video track's codec name is different
+    // or if no local current stream
+    // or if audio/subtitle stream does exists anymore
```
I don't understand those 3 line comments well

Commit:
add ffmpeg "blocksize" option

-> it aparently improves stability

Commit:
expose ffmpeg "blocksize" option

value must be around 200k on windows, but 800k on mac

Commit:
small lint fix

Commit:
Merge branch 'base' into feature/FLOW-3451-aes-decryption
